server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/sql/RenameColumnsBuilder.java

```diff
@@ -0,0 +1,136 @@
+import java.util.stream.Collectors;
+import javax.annotation.CheckForNull;
+import org.sonar.db.dialect.Dialect;
+import org.sonar.db.dialect.H2;
+import org.sonar.db.dialect.MsSql;
+import org.sonar.db.dialect.MySql;
+import org.sonar.db.dialect.Oracle;
+import org.sonar.db.dialect.PostgreSql;
+import org.sonar.server.platform.db.migration.def.ColumnDef;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Lists.newArrayList;
+import static org.sonar.server.platform.db.migration.def.Validations.validateColumnName;
+import static org.sonar.server.platform.db.migration.def.Validations.validateTableName;
+
+/**
+ * This builder have the main goal to change the name column.
+ *
+ * For MySQL the column definition is mandatory, however the change of column
+ * is not supported on this class.
+ * In case of renaming and changing a column, this must be done in two separate steps,
+ * first rename the column then update the types of this column with @see {@link AlterColumnsBuilder}
+ */
+public class RenameColumnsBuilder {
+
+  private final Dialect dialect;
+  private final String tableName;
+  private final List<RenameColumnDef> renameColumnDefs = newArrayList();
+
+  public RenameColumnsBuilder(Dialect dialect, String tableName) {
+    this.dialect = dialect;
+    this.tableName = tableName;
+  }
+
+  public RenameColumnsBuilder renameColumn(String newColumnName, ColumnDef columnDef) {
+    renameColumnDefs.add(new RenameColumnDef(newColumnName, columnDef));
+    return this;
+  }
+
+  public List<String> build() {
+    validateTableName(tableName);
+    renameColumnDefs.stream().forEach(
+      r -> {
+        validateColumnName(r.getNewColumnName());
+        validateColumnName(r.getOldColumnName());
+        checkArgument(!r.getNewColumnName().equals(r.getOldColumnName()), "Column names must be different");
+      }
+    );
+    return createSqlStatement();
+  }
+
+  private List<String> createSqlStatement() {
+    return renameColumnDefs.stream().map(
+      r -> {
+        switch (dialect.getId()) {
+          case H2.ID:
+            return "ALTER TABLE " + tableName + " ALTER COLUMN " + r.getOldColumnName() + " RENAME TO " + r.getNewColumnName();
+          case Oracle.ID:
+          case PostgreSql.ID:
+            return "ALTER TABLE " + tableName + " RENAME COLUMN " + r.getOldColumnName() + " TO " + r.getNewColumnName();
+          case MySql.ID:
+            return "ALTER TABLE " + tableName + " CHANGE " + r.getOldColumnName() + " TO " + r.getNewColumnName() + " " + r.generateSqlType(dialect);
+          case MsSql.ID:
+            return "EXEC sp_rename '" + tableName + "." + r.getOldColumnName() + "', '" + r.getNewColumnName() + "', 'COLUMN'";
+          default:
+            throw new IllegalArgumentException("Unsupported dialect id " + dialect.getId());
+        }
+      }
+    ).collect(Collectors.toList());
+  }
+
+  private class RenameColumnDef implements ColumnDef {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this a "static" inner class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)